postgres/src/main/scala/zio/sql/postgresql/PostgresModule.scala

```diff
@@ -1,5 +1,205 @@
+      case Expr.Source(tableName, column)                               =>
+        val _ = builder.append(tableName).append(".").append(column.name)
+      case Expr.Unary(base, op)                                         =>
+        val _ = builder.append(" ").append(op.symbol)
+        buildExpr(base)
+      case Expr.Property(base, op)                                      =>
+        buildExpr(base)
+        val _ = builder.append(" ").append(op.symbol)
+      case Expr.Binary(left, right, op)                                 =>
+        buildExpr(left)
+        builder.append(" ").append(op.symbol).append(" ")
+        buildExpr(right)
+      case Expr.Relational(left, right, op)                             =>
+        buildExpr(left)
+        builder.append(" ").append(op.symbol).append(" ")
+        buildExpr(right)
+      case Expr.In(value, set)                                          =>
+        buildExpr(value)
+        buildReadString(set)
+      case Expr.Literal(value)                                          =>
+        val _ = builder.append(value.toString) //todo fix escaping
+      case Expr.AggregationCall(param, aggregation)                     =>
+        builder.append(aggregation.name.name)
+        builder.append("(")
+        buildExpr(param)
+        val _ = builder.append(")")
+      case Expr.FunctionCall1(param, function)                          =>
+        builder.append(function.name.name)
+        builder.append("(")
+        buildExpr(param)
+        val _ = builder.append(")")
+      case Expr.FunctionCall2(param1, param2, function)                 =>
+        builder.append(function.name.name)
+        builder.append("(")
+        buildExpr(param1)
+        builder.append(",")
+        buildExpr(param2)
+        val _ = builder.append(")")
+      case Expr.FunctionCall3(param1, param2, param3, function)         =>
+        builder.append(function.name.name)
+        builder.append("(")
+        buildExpr(param1)
+        builder.append(",")
+        buildExpr(param2)
+        builder.append(",")
+        buildExpr(param3)
+        val _ = builder.append(")")
+      case Expr.FunctionCall4(param1, param2, param3, param4, function) =>
+        builder.append(function.name.name)
+        builder.append("(")
+        buildExpr(param1)
+        builder.append(",")
+        buildExpr(param2)
+        builder.append(",")
+        buildExpr(param3)
+        builder.append(",")
+        buildExpr(param4)
+        val _ = builder.append(")")
+    }
+
+    def buildReadString[A <: SelectionSet[_]](read: self.Read[_]): Unit =
+      read match {
+        //todo offset (needs orderBy, must use fetch _instead_ of top)
+        case Read.Select(selection, table, whereExpr, groupBy, havingExpr, orderBy, offset, limit) =>
+          builder.append("select ")
+          limit match {
+            case Some(limit) =>
+              builder.append("top ").append(limit).append(" ")
```

Comment:
I believe top is a sql server only special. This whole function should be reviewed to see if there's any other sql server/postgres differences. for postgres I believe for postgres this needs to be "limit x" at the end of the statement. the offset issue from the comment should be a non-issue for postgres.

Reply:
Sorry, missed that. And yeah, that's actually one of the reasons we need to have these renderers separate.